src/index.js

```diff
@@ -708,7 +696,34 @@ class NearProvider {
                     GAS_AMOUNT,
                     val
                 );
+            } else if (to != `0x${"0".repeat(40)}` && to == txObj.from) {
+                // Add near
+                console.log("**ADD NEAR**")
```

Comment:
we need to add near before we send funds to another evm address. Could this be done in one transaction?

Reply:
Can you elaborate on this? Who are the sender(s) and receiver(s) in this case?

Reply:
the way near-evm is written there are two distinct functions:
* `add_near` which transfers ownership of $near from a near-account to the associated eth-account.
* `move_funds_to_evm_address` allows an eth-account to transfer funds to another eth-account (once they have ownership of $near)

Asking for any thoughts on whether it's a good idea to be able to expedite this into one transaction to avoid the tedium, or whether this is intended as a potential safeguard to the user.

Reply:
or maybe it's just impossible.

Reply:
You cannot batch two function calls with different sender and receiver in the same transaction.